utils/nsynth_dataset_group.py

```diff
@@ -0,0 +1,113 @@
+"""Program groups and compresses files from NSynth Dataset.
+
+NSynth Dataset (https://magenta.tensorflow.org/datasets/nsynth)
+consists of .wav files containing various instruments' sounds.
+This program groups those files into separate folders based on
+the information about instrument category present in the filename
+and compresses those folders into .tar.gz files.
+"""
+
+import argparse
+import os
+import re
+import tarfile
+
+
+def group_files(input_directory, temporary_directory):
+    """Groups files into separate directories.
+
+    Groups .wav files from the input directory into separate folders
+    based on the instrument category present in the filename.
+
+    Args:
+        input_directory:
+          A directory where mixed audio files are located.
+        temporary_directory:
+          A base directory where separate instruments' folders will be stored.
+    """
+    instruments = set()
+
+    for filename in os.listdir(input_directory):
+        # Filename is concatenation of instrument category, example number
+        # and audio file extension, for example: instrument_category_012-345-678.wav.
+        # We only want to extract the instrument category.
+        if re.fullmatch(r'.+_[0-9\-]{11}\.wav', filename):
```

Comment:
This regexp match seems to be very "adhoc". I would move this outside of the function and make it a constant.

Reply:
Done